addons/hw_drivers/drivers/PrinterDriver.py

```diff
@@ -0,0 +1,302 @@
+        """Updates the state of the current printer.
+
+        Args:
+            status (str): The new value of the status
+            message (str): A comprehensive message describing the status
+            reason (str): The reason fo the current status
+        """
+        if self.state['status'] != status or self.state['reason'] != reason:
+            self.state = {
+                'status': status,
+                'message': message,
+                'reason': reason,
+            }
+            self.send_status()
+
+    def send_status(self):
+        """ Sends the current status of the printer to the connected Odoo instance.
+        """
+        self.data = {
+            'value': '',
+            'state': self.state,
+        }
+        event_manager.device_changed(self)
+
+    def print_raw(self, data):
+        process = subprocess.Popen(["lp", "-d", self.device_identifier], stdin=subprocess.PIPE)
+        process.communicate(data)
+
+    def print_receipt(self, receipt):
+        im = Image.open(io.BytesIO(receipt))
+
+        # Convert to greyscale then to black and white
+        im = im.convert("L")
+        im = ImageOps.invert(im)
+        im = im.convert("1")
+
+        '''ESC a n
+            - ESC a: Select justification, in hex: "1B 61"
+            - n: Justification:
+                - 0: left
+                - 1: center
+                - 2: right
+        '''
+        center = b'\x1b\x61\x01'
+
+        '''GS v0 m x y
+            - GS v0: Print raster bit image, in hex: "1D 76 30"
+            - m: Density mode:
+                - 0: 180dpi x 180dpi
+                - 1: 180dpi x 90dpi
+                - 2: 90dpi x 180 dpi
+                - 3: 90dpi x 90 dpi
+            - x: Length in X direction, in bytes, represented as 2 bytes in little endian
+            - y: Length in Y direction, in dots, represented as 2 bytes in little endian
+        '''
+        width_pixels, height_pixels = im.size
+        width_bytes = int((width_pixels + 7) / 8)
+        print_command = b"\x1d\x76\x30" + b'\x00' + (width_bytes).to_bytes(2, 'little') + height_pixels.to_bytes(2, 'little')
+
+        '''GS V m
+            - GS V: Cut, in hex: "1D 56"
+            - m: Cut mode:
+                - 0: Full cut
+                - 1: Partial cut
+                - 65 (0x41): Feed paper then full cut
+                - 66 (0x42): Feed paper then partial cut
+        '''
+        cut = b'\x1d\x56' + b'\x41'
+
+        self.print_raw(center + print_command + im.tobytes() + cut + b'\n')
+
+    def print_status(self):
+        """Prints the status ticket of the IoTBox on the current printer."""
+        wlan = ''
+        ip = ''
+        mac = ''
+        homepage = ''
+
+        hosting_ap = os.system('pgrep hostapd') == 0
+        ssid = subprocess.check_output('iwconfig 2>&1 | grep \'ESSID:"\' | sed \'s/.*"\\(.*\\)"/\\1/\'', shell=True).decode('utf-8').rstrip()
+        if hosting_ap:
+            wlan = 'Wireless network:\nIoTBox\n\n'
```

Comment:
check name for SSID in this file `/root_bypass_ramdisks/etc/hostapd/hostapd.conf`